python/tank/log.py

```diff
@@ -235,6 +237,125 @@ class LogManager(object):
+        rename files will be ignored and the handler will keep writing to the current file. A message
+        will also be logged at the exception level so the user is aware that something really bad just
+        happened. Finally, the handler will not try to rollover in the future.
+        """
+
+        def __init__(self, filename, mode="a", maxBytes=0, backupCount=0, encoding=None, delay=0):
+            """
+            :param str filename: Name of of the log file.
+            :param str mode: Mode to open the file, should be  "w" or "a". Defaults to "a"
+            :param int maxBytes: Maximum file size before rollover. By default, rollover never happens.
+            :param int backupCount: Number of backups to make. Defaults to 0.
+            :param encoding: Encoding to use when writing to the file. Defaults to None.
+            :param delay: If the value is truthy, the log file will be opened only on first write.
+                File will be opened by default.
+            """
+            super(LogManager._SafeRotatingFileHandler, self).__init__(filename, mode, maxBytes, backupCount, encoding, delay)
+            self._disable_rollover = False
+
+        def doRollover(self):
+            """
+            Rename every backups so the current log can be promoted to backup number one.
+
+            The new log file is empty. If this process fails due to any I/O error, rollover is
+            deactivated for this handler and logs will be appended to the current log file indefinitely.
+            """
+
+            temp_backup_name = "%s.%s" % (self.baseFilename, uuid.uuid4())
+
+            # We need to close the file before renaming it (windows!)
+            if self.stream:
+                self.stream.close()
+                self.stream = None
+
+            # Before doing the rollover, check if the first file will fail at all.
+            # If it does, then it is a good thing that we checked otherwise the last
+            # backup would have been blown away before encountering the error.
+
+            # Take the scenario where there's only one backup. This means that
+            # doRollover would first delete the backup (.1) file so it can make
+            # room for the main file to be renamed to .1. However, if the main file
+            # can't be renamed, we've effectively lost 50% of the logs we had, which
+            # is not cool. Since most of the time only the first file will be locked,
+            # we will try to rename it first. If that fails right away as expected,
+            # we don't try any rollover and append to the current log file.
+            # and raise the _disable_rollover flag.
+
+            try:
+                os.rename(self.baseFilename, temp_backup_name)
+            except:
+                # It failed, so we'll simply append from now on.
+                log.debug("Can't rename main file:", exc_info=True)
```

Comment:
i think it would be helpful to say something like

"Cannot rotate log file '%s'. Will continue to log to this file, exceeding the specified maximum size"